src/fides/api/models/privacy_notice.py

```diff
@@ -332,123 +365,249 @@ def create(
         data: dict[str, Any],
         check_name: bool = False,
     ) -> PrivacyNotice:
+        """
+        Creates a Privacy Notice and then a NoticeTranslation record for each supplied Translation.
+
+        - For each Notice translation, we also create a PrivacyNoticeHistory record which versions details
+        from both the Notice and the Translation itself for auditing purposes.
+        - Privacy preferences are saved against the PrivacyNoticeHistory record which contains all the info
+        about the notice and the translation to which the user supplied a consent preference.
+        """
+        translations = data.pop("translations", []) or []
         created = super().create(db=db, data=data, check_name=check_name)
+        data.pop(
+            "id", None
+        )  # Default Notices have id specified but we don't want to use the same id for the historical record
+
+        for translation_data in translations:
+            # Create the Notice Translation
+            translation = NoticeTranslation.create(
+                db, data={**translation_data, "privacy_notice_id": created.id}
+            )
+            # Take a snapshot of the Privacy Notice and the Notice Translation combined. Privacy preferences
+            # are saved against this record.
+            PrivacyNoticeHistory.create(
+                db,
+                data={**data, **translation_data, "translation_id": translation.id},
+                check_name=False,
+            )
 
-        # create the history after the initial object creation succeeds, to avoid
-        # writing history if the creation fails and so that we can get the generated ID
-        data.pop("id", None)
-        history_data = {**data, "privacy_notice_id": created.id}
-        PrivacyNoticeHistory.create(db, data=history_data, check_name=False)
         return created
 
     def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyNotice:
```

Comment:
This is a core method of this PR that received major changes, PrivacyNotice.update